src/ts/components/subtitleoverlay.ts

```diff
@@ -416,3 +436,98 @@ class ActiveSubtitleManager {
     this.activeSubtitleCueCount = 0;
   }
 }
+
+export class SubtitleRegionContainerManager {
+  private subtitleRegionContainers: { [regionName: string]: SubtitleRegionContainer } = {};
+
+  /**
+   * @param subtitleOverlay Reference to the subtitle overlay for adding and removing the containers.
+   */
+  constructor(private subtitleOverlay: SubtitleOverlay) {
+    this.subtitleOverlay = subtitleOverlay;
+  }
+
+  /**
+   * Creates and wraps a subtitle label into a container div based on the subtitle region.
+   * If the subtitle has positioning information it is added to the container.
+   * @param label The subtitle label to wrap
+   */
+  addLabel(label: SubtitleLabel): void {
+    const regionName = label.region || 'default';
+    if (!this.subtitleRegionContainers[regionName]) {
+      const regionContainer = new SubtitleRegionContainer({
+        cssClass: `subtitle-position-${regionName}`,
+      });
+
+      this.subtitleRegionContainers[regionName] = regionContainer;
+
+      if (label.regionStyle) {
+        regionContainer.getDomElement().attr('style', label.regionStyle);
+      } else {
+        // getDomElement needs to be called at least once to ensure the component exists
+        regionContainer.getDomElement();
+      }
+
+      for (const regionName in this.subtitleRegionContainers) {
+        this.subtitleOverlay.addComponent(this.subtitleRegionContainers[regionName]);
+      }
+    }
+
+    this.subtitleRegionContainers[regionName].addLabel(label);
+  }
+
+  /**
+   * Removes a subtitle label from a container.
+   */
+  removeLabel(label: SubtitleLabel): void {
+    const region = label.region || 'default';
+    for (const regionName in this.subtitleRegionContainers) {
+      this.subtitleRegionContainers[regionName].removeLabel(label);
+    }
+
+    // Remove container if no more labels are displayed
+    if (this.subtitleRegionContainers[region].isEmpty()) {
+      this.subtitleOverlay.removeComponent(this.subtitleRegionContainers[region]);
+      delete this.subtitleRegionContainers[region];
+    }
+  }
+
+  /**
+   * Removes all subtitle containers.
+   */
+  clear(): void {
+    for (const regionName in this.subtitleRegionContainers) {
+      this.subtitleOverlay.removeComponent(this.subtitleRegionContainers[regionName]);
+    }
+
+    this.subtitleRegionContainers = {};
+  }
+}
+
+class SubtitleRegionContainer extends Container<ContainerConfig> {
+  private labelCount = 0;
+
+  constructor(config: ContainerConfig = {}) {
+    super(config);
+
+    this.config = this.mergeConfig(config, {
+      cssClasses: ['subtitle-region-container'],
+    }, this.config);
+  }
+
+  addLabel(labelToAdd: SubtitleLabel) {
+    this.labelCount++;
+    this.addComponent(labelToAdd);
+    this.updateComponents();
+  }
+
+  removeLabel(labelToRemove: SubtitleLabel): void {
+    this.labelCount--;
+    this.removeComponent(labelToRemove);
+    this.updateComponents();
+  }
+
+  public isEmpty(): boolean {
+    return this.labelCount === 0;
```

Comment:
You could omit the `labelCount` and check `this.getComponents().length` instead.

Reply:
I agree, but unfortunately that breaks the unit tests as I have to mock the container class.

Reply:
Ok, I think we can live with that :)